docs/api.md

```diff
@@ -30,6 +30,21 @@ var html = katex.renderToString("c = \\pm\\sqrt{a^2 + b^2}", {
 // '<span class="katex">...</span>'
 ```
 
+## Persistent Macros
+
+KaTeX documentation tells the author that `\gdef` will create a macro that persists between KaTeX fields. In order to fulfill that promise, you must make certain provisions. Say that a parser has dropped from the sky with an array of DOM elements that contain math. You could then write this code:
+
+```js
+const macros = {};
+for (let element of mathElements) {
+    katex.render(element.textContent, element, {
+        throwOnError: false,
+        macros
+    };
+```
+
+Notice that you create the `macros` object outside the loop. That object will contain macros that the user defines using `\gdef` and the macros will persist between calls to `katex.render`.
+
```

Comment:
It's probably worth warning about security:

```suggestion

### Security of Persistent Macros
Persistent macros can change the behavior of KaTeX (e.g. redefining standard commands), so for security, such a setup should be used only for multiple elements of common trust.  For example, you might enable persistent macros within a message posted by a single user (by creating a `macros` object for that message), but you probably should not enable persistent macros across multiple messages posted by multiple users.
```

Reply:
Good point. I'll add it.